### PR TITLE
feat: update helm chart to support aws

### PR DESCRIPTION
**Pull request by saulfrance**
#### 📲 What

Update the Helm chart to support AWS deployments

#### 🤔 Why
		
Currently we use different methods for Azure and AWS deployments to Kubernetes. Helm charts are dynamic and can support both.
		
#### 🛠 How
		
- Update the Helm chart for dynamic configuration based on cloud provider.
- Switch deploy task to use helm
- Update pipeline to set Helm variables

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: deploy/helm/stacks-dotnet/templates/secret.yaml [hunk 1]
```diff
@@ -9,4 +9,10 @@ metadata:
   namespace: {{ .Values.namespace }}
 type: Opaque
 data:
+  {{- if eq .Values.cloudProvider "azure" }}
   app_insights: {{ .Values.pod.appInsightsKey | b64enc | quote }}
+  {{- end }}
+  {{- if eq .Values.cloudProvider "aws" }}
+  CLOUDWATCH_LOG_GROUP: {{ .Values.pod.cloudwatchLogGroup | b64enc | quote  }}
+  CLOUDWATCH_STREAM_PREFIX: {{.Values.pod.cloudwatchStreamPrefix | b64enc | quote }}
```
Comment (ElvenSpellmaker) [inline deploy/helm/stacks-dotnet/templates/secret.yaml:17]: Is the app currently logging to CW directly? We have Container Insights installed on the cluster which ships all stdout to CW automatically, unless these are extra metrics or something

Comment (saulfrance) [reply deploy/helm/stacks-dotnet/templates/secret.yaml:17]: Doesn't look like it is used on the app from searching for those environment variables. I was just converting what was already in the Kubernetes manifests so didn't check. Would you expect to find the logs in this group /aws/eks/ensono-stacks-nonprod-ew2-eks/cluster? I can't see any for the dotnet api

Comment (ElvenSpellmaker) [reply deploy/helm/stacks-dotnet/templates/secret.yaml:17]: They're under `/aws/eks/ensono-stacks-nonprod-ew2-eks/application`
image

> I was just converting what was already in the Kubernetes manifests so didn't check

Yeah no worries, those manifests are all very old, as they were before we moved the cluster to the new one, which has Enhanced CI installed and nginx (hence the AWS LB Controller annotations in the Ingress that you removed)

I'm wondering if they're even needed any more, as we ship all stdout?

Comment (saulfrance) [reply deploy/helm/stacks-dotnet/templates/secret.yaml:17]: Fair enough, I'll remove them a check we're still getting logs. We can always add them back if needed

Comment (ElvenSpellmaker) [reply deploy/helm/stacks-dotnet/templates/secret.yaml:17]: Cool cool

File: deploy/helm/stacks-dotnet/values.yaml [hunk 2]
```diff
@@ -32,6 +33,8 @@ aadPodIdentityBinding: ${aadpodidentitybinding}
 pod:
   secretSuffix: secret
   appInsightsKey: ${app_insights_key}
+  cloudwatchLogGroup: ${cloudwatch_log_group}
+  cloudwatchStreamPrefix: ${cloudwatch_stream_prefix}
```
Comment (ElvenSpellmaker) [inline deploy/helm/stacks-dotnet/values.yaml:37]: Can these go?

Comment (saulfrance) [reply deploy/helm/stacks-dotnet/values.yaml]: Removed

File: deploy/helm/stacks-dotnet/templates/secret.yaml (hunk 1 above)
Comment (ElvenSpellmaker) [reply deploy/helm/stacks-dotnet/templates/secret.yaml]: Nice nice, thanks for confirming 👍🏻

File: deploy/helm/stacks-dotnet/values.yaml [hunk 3]
```diff
@@ -82,7 +83,7 @@ affinity: {}
 
 app:
   project: ${project}
-  company: {$company}
+  company: ${company}
```
Comment (ElvenSpellmaker) [inline deploy/helm/stacks-dotnet/values.yaml:86]: lolol

**Comment by ElvenSpellmaker** [comment]
Nice one mate, thanks for this

**Review by ElvenSpellmaker — APPROVED** [review]

**Pull request merged**